File: repos_estudiantes_TARQUI_tercera_entrega/SIÑANI_QUELALI_IVAN/tarea3/ge4.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
public class Vector<T extends Comparable<T>> {
    private List<T> elementos = new ArrayList<>();
    public void agregar(T elemento) {
        elementos.add(elemento);
    }
    public T obtener(int i) {
        if (i >= 0 && i < elementos.size()) {
            return elementos.get(i);
        } else {
            return null; 
        }
    }
    public T obtenerMayor() {
        return Collections.max(elementos);
    }
    public T obtenerMenor() {
        return Collections.min(elementos);
    }
}
public class Main {
    public static void main(String[] args) {
        Vector<Integer> vector = new Vector<>();
        vector.agregar(10);
        vector.agregar(5);
        vector.agregar(20);
        
        System.out.println("Elemento en posición 1: " + vector.obtener(1));
        System.out.println("Valor mayor: " + vector.obtenerMayor());
        System.out.println("Valor menor: " + vector.obtenerMenor());
    }
}
//DIAGRAMA
//Clase Vector<T extends Comparable<T>>
//-------------------------------------
//- elementos: List<T>
//-------------------------------------
//+ agregar(T elemento): void
//+ obtener(int i): T
//+ obtenerMayor(): T
//+ obtenerMenor(): T
//-------------------------------------
//Clase Main
//-------------------------------------
//+ main(String[] args): void
//-------------------------------------

File: repos_estudiantes_TARQUI_tercera_entrega/SIÑANI_QUELALI_IVAN/tarea3/ge4.py
class Vector:
    def __init__(self):
        self.elementos = []

    def agregar(self, elemento):
        self.elementos.append(elemento)

    def obtener(self, i):
        return self.elementos[i] if 0 <= i < len(self.elementos) else None

    def obtener_mayor(self):
        return max(self.elementos) if self.elementos else None

    def obtener_menor(self):
        return min(self.elementos) if self.elementos else None

vector = Vector()
vector.agregar(10)
vector.agregar(5)
vector.agregar(20)

print("Elemento en posición 1:", vector.obtener(1))
print("Valor mayor:", vector.obtener_mayor())
print("Valor menor:", vector.obtener_menor())
